template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/kudu/KuduMasterConfigProvider.java

```diff
@@ -11,19 +11,19 @@
 
 import com.cloudera.api.swagger.model.ApiClusterTemplateConfig;
 import com.google.api.client.util.Lists;
-import com.sequenceiq.cloudbreak.cmtemplate.CmTemplateComponentConfigProvider;
-import com.sequenceiq.cloudbreak.cmtemplate.CmTemplateProcessor;
+import com.sequenceiq.cloudbreak.cmtemplate.configproviders.AbstractRoleConfigProvider;
 import com.sequenceiq.cloudbreak.template.TemplatePreparationObject;
 
 @Component
-public class KuduServiceConfigProvider implements CmTemplateComponentConfigProvider {
+public class KuduMasterConfigProvider extends AbstractRoleConfigProvider {
 
     @Override
-    public List<ApiClusterTemplateConfig> getServiceConfigs(CmTemplateProcessor templateProcessor, TemplatePreparationObject source) {
+    public List<ApiClusterTemplateConfig> getRoleConfigs(String roleType, TemplatePreparationObject source) {
         ArrayList<ApiClusterTemplateConfig> configs = Lists.newArrayList();
 
-        String cdhVersion = templateProcessor.getStackVersion();
-        if (isVersionNewerOrEqualThanLimited(cdhVersion, CLOUDERA_STACK_VERSION_7_2_11)) {
+        String cdhVersion = source.getBlueprintView().getProcessor().getStackVersion() == null ?
+            "" : source.getBlueprintView().getProcessor().getStackVersion();
+        if (isVersionNewerOrEqualThanLimited(cdhVersion, CLOUDERA_STACK_VERSION_7_2_11) && roleType == "KUDU_MASTER") {
```

Comment:
Assuming you indeed need a role config (and not a service-level one): The extra term `roleType == "KUDU_MASTER"` is redundant because of this below:
```
    @Override
    public List<String> getRoleTypes() {
        return List.of(KuduRoles.KUDU_MASTER);
    }
```
So this config provider will only be evaluated for that single role type anyway.